cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/logging/Log.java

```diff
@@ -66,20 +84,65 @@
         }
     }
 
+    private Log(String name) {
+        logger = Logger.getLogger(name);
+
+        // we want to enable really fine logs on this logger by default
+        // log level filtering is done in the baseLogger
+        logger.setLevel(DEBUG_LEVEL);
+    }
+
+    /**
+     * Factory method. Returns log object for a given name. New log objects will be created on demand.
+     */
+    public static Log getLog(String name) {
+        final String baseLoggerName = baseLogger.getName();
+
+        // if we permitted such loggers, the handler registration etc. wouldn't work
+        // they all expect a constant prefix
+        if (!name.startsWith(baseLoggerName)) {
+            throw new RuntimeException("logger name does not start with global prefix " + baseLoggerName);
```

Comment:
In my opinion an IllegalArgumentException would fit better

Reply:
I'm not a Java pro, all I wanted to have is an unchecked exception. Your suggestion obviously makes more sense and is derived from `RuntimeException`, too.